### PR TITLE
Travis CI: Update HHVM and use Ubuntu 14.04 Trusy

### PR DESCRIPTION
**Pull request by ntwb**
#### Changes proposed in this Pull Request:
- This pull request updates the HHVM version to the latest v3.13.1

See the equivalent WordPress core ticket [#WP36930](https://core.trac.wordpress.org/ticket/36930)

-------------------
- [x] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).

The patch has some extra churn as indentation was using 3 spaces rather than 2 in places 😱

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,13 +3,10 @@
 # Tell Travis CI we're using PHP
 language: php
 
-# PHP version used in first build configuration.
-php:
-    - "5.6"
```
Comment (kraftbj) [inline .travis.yml]: Could you add in a 5.6/master in the matrix below to make up for this removal?

**Comment by kraftbj** [comment]
Besides the 5.6/master loss, I like it (and was tracking the core ticket to mimic what you were doing there too). Thanks!

**Comment by ntwb** [comment]
> Besides the 5.6/master loss, I like it (and was tracking the core ticket to mimic what you were doing there too). Thanks!

Yeah, I saw you were watching the ticket so I came over here :)

File: .travis.yml (hunk 1 above)
Comment (ntwb) [inline .travis.yml]: Yep, will do :+1

**Comment by ntwb** [comment]
The current HHVM job is "stuck", it looks like it is erroring, why, I've no idea 😁 

https://travis-ci.org/Automattic/jetpack/jobs/132733960#L2901

**Comment by ntwb** [comment]
The most recent Travis CI build from this PR: https://travis-ci.org/Automattic/jetpack/builds/132736293

The HHVM Job of that build: https://travis-ci.org/Automattic/jetpack/jobs/132736307

**Comment by kraftbj** [comment]
Looks like that one finally finished. I restarted a couple of jobs on the previous build ( https://travis-ci.org/Automattic/jetpack/builds/132733947 ). We're doing some new stuff with npm and Travis has been occasionally erroring out on `npm install`. Haven't dug too much into that yet; waiting for development to stabilize a bit.

Heading off for the night, but assuming nothing unexpected comes off of the builds in progress, we'll get this in. Thanks again!

**Comment by ntwb** [comment]
The Travis CI build has now finished https://travis-ci.org/Automattic/jetpack/builds/132736293

Every single job, yes, **every** job has a failing PHPUnit test that is failing and Travis CI is not returning the failing status, see https://github.com/Automattic/jetpack/issues/3955

**Comment by kraftbj** [comment]
With the Travis/Gulp issues not withstanding, this looks good for HHVM. Thanks!

**Pull request merged**